diff -urN libpng-1.2.0.org/scripts/makefile.gcmmx libpng-1.0.9/scripts/makefile.gcmmx
--- libpng-1.2.0.org/scripts/makefile.gcmmx	Thu Feb  1 15:22:46 2001
+++ libpng-1.2.0/scripts/makefile.gcmmx	Thu Feb  1 15:26:38 2001
@@ -17,8 +17,8 @@
 # Where the zlib library and include files are located
 #ZLIBLIB=/usr/local/lib
 #ZLIBINC=/usr/local/include
-ZLIBLIB=../zlib
-ZLIBINC=../zlib
+ZLIBLIB=/usr/lib
+ZLIBINC=
 
 ALIGN=
 # for i386:
@@ -31,8 +31,8 @@

 # Remove -DPNG_THREAD_UNSAFE_OK if you need thread safety
 ### for generic gcc:
-CFLAGS=-DPNG_THREAD_UNSAFE_OK -DPNG_USE_PNGGCCRD -I$(ZLIBINC) -Wall \
-       	-O3 $(ALIGN) -funroll-loops \
+CFLAGS=-DPNG_THREAD_UNSAFE_OK -DPNG_USE_PNGGCCRD -I$(ZLIBINC) -Wall \
+	$(OPT_FLAGS) $(ALIGN) -funroll-loops \
 	-fomit-frame-pointer  # $(WARNMORE) -g -DPNG_DEBUG=5
 ### for gcc 2.95.2 on 686:
 #CFLAGS=-DPNG_THREAD_UNSAFE_OK -DPNG_USE_PNGGCCRD -I$(ZLIBINC) -Wall -O3 \
diff -urN libpng-1.0.9.org/scripts/makefile.linux libpng-1.0.9/scripts/makefile.linux
--- libpng-1.0.9.org/scripts/makefile.linux	Thu Feb  1 15:22:46 2001
+++ libpng-1.0.9/scripts/makefile.linux	Thu Feb  1 15:25:49 2001
@@ -11,8 +11,8 @@
 # Where the zlib library and include files are located
 #ZLIBLIB=/usr/local/lib
 #ZLIBINC=/usr/local/include
-ZLIBLIB=../zlib
-ZLIBINC=../zlib
+ZLIBLIB=/usr/lib
+ZLIBINC=
 
 ALIGN=
 # for i386:
@@ -24,7 +24,7 @@
 
 # for pgcc version 2.95.1, -O3 is buggy; don't use it.
 
-CFLAGS=-I$(ZLIBINC) -Wall -O3 -funroll-loops \
+CFLAGS=-I$(ZLIBINC) -Wall $(OPT_FLAGS) -funroll-loops \
 	$(ALIGN) # $(WARNMORE) -g -DPNG_DEBUG=5
 LDFLAGS=-L. -Wl,-rpath,. -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) -lpng -lz -lm
 
