--- libpng-1.0.8/contrib/pngminus/makefile.std.old	Sat Nov 27 13:41:26 1999
+++ libpng-1.0.8/contrib/pngminus/makefile.std	Wed Nov  1 13:05:06 2000
@@ -9,18 +9,18 @@
 RM=rm
 CP=cp
 
-PNGPATH = /usr/local
+PNGPATH = ../..
 PNGINC = $(PNGPATH)/include
-#PNGLIB = $(PNGPATH)/lib -lpng
-PNGLIB = $(PNGPATH)/lib/libpng.a
+PNGLIB = $(PNGPATH)/ -lpng
+#PNGLIB = $(PNGPATH)/lib/libpng.a
 
 ZPATH = /usr/local
 ZINC = $(ZPATH)/include
 #ZLIB = $(ZPATH)/lib -lz
 ZLIB = $(ZPATH)/lib/libz.a
 
-CCFLAGS=-I$(PNGINC) -I$(ZINC)
-LDFLAGS=-L$(PNGLIB) -L$(ZLIB)
+CCFLAGS=$(RPM_OPT_FLAGS) -I$(PNGINC)
+LDFLAGS= -s -L$(PNGLIB)
 C=.c
 O=.o
 L=.a
