--- libpng-1.0.14/contrib/pngminus/makefile.std.orig	Mon Jul  8 02:47:58 2002
+++ libpng-1.0.14/contrib/pngminus/makefile.std	Thu Jul 18 20:03:13 2002
@@ -19,11 +19,11 @@
 #ZINC = -I$(ZPATH)/include
 #ZLIB = -L$(ZPATH)/lib -lz
 #ZLIBS = $(ZPATH)/lib/libz.a
-ZINC = -I../../../zlib
-ZLIB = -L../../../zlib -lz
-ZLIBS = ../../../zlib/libz.a
+ZINC =
+ZLIB = -lz
+ZLIBS = -lz
 
-CFLAGS=-O3 $(PNGINC) $(ZINC)
+CFLAGS=$(OPT_FLAGS) $(PNGINC) $(ZINC)
 LDFLAGS=$(PNGLIB) $(ZLIB)
 LDFLAGSS=$(PNGLIBS) $(ZLIBS)
 C=.c
