--- libpng-1.2.38/png.h.orig	2009-07-16 12:58:31.000000000 +0200
+++ libpng-1.2.38/png.h	2009-07-19 18:19:09.000000000 +0200
@@ -1842,16 +1842,16 @@
 extern PNG_EXPORT(void,png_destroy_read_struct) PNGARG((png_structpp
    png_ptr_ptr, png_infopp info_ptr_ptr, png_infopp end_info_ptr_ptr));
 
-/* Free all memory used by the read (old method - NOT DLL EXPORTED) */
-extern void png_read_destroy PNGARG((png_structp png_ptr, png_infop info_ptr,
-   png_infop end_info_ptr));
+/* free all memory used by the read (old method) */
+extern PNG_EXPORT(void,png_read_destroy) PNGARG((png_structp png_ptr,
+   png_infop info_ptr, png_infop end_info_ptr));
 
 /* Free any memory associated with the png_struct and the png_info_structs */
 extern PNG_EXPORT(void,png_destroy_write_struct)
    PNGARG((png_structpp png_ptr_ptr, png_infopp info_ptr_ptr));
 
-/* Free any memory used in png_ptr struct (old method - NOT DLL EXPORTED) */
-extern void png_write_destroy PNGARG((png_structp png_ptr));
+/* Free any memory used in png_ptr struct (old method) */
+extern PNG_EXPORT(void,png_write_destroy) PNGARG((png_structp png_ptr));
 
 /* Set the libpng method of handling chunk CRC errors */
 extern PNG_EXPORT(void,png_set_crc_action) PNGARG((png_structp png_ptr,
--- libpng-1.2.37/pngread.c~	2009-06-04 13:08:00.000000000 +0200
+++ libpng-1.2.37/pngread.c	2009-06-18 10:49:04.000000000 +0200
@@ -1157,7 +1157,7 @@
 }
 
 /* Free all memory used by the read (old method) */
-void /* PRIVATE */
+void PNGAPI
 png_read_destroy(png_structp png_ptr, png_infop info_ptr, png_infop end_info_ptr)
 {
 #ifdef PNG_SETJMP_SUPPORTED
--- libpng-1.2.37/pngwrite.c~	2009-06-04 13:08:00.000000000 +0200
+++ libpng-1.2.37/pngwrite.c	2009-06-18 10:50:07.000000000 +0200
@@ -1082,7 +1082,7 @@
 
 
 /* Free any memory used in png_ptr struct (old method) */
-void /* PRIVATE */
+void PNGAPI
 png_write_destroy(png_structp png_ptr)
 {
 #ifdef PNG_SETJMP_SUPPORTED
