diff -urN libpng-1.2.1.org/contrib/pngminus/makefile.std libpng-1.2.1/contrib/pngminus/makefile.std
--- libpng-1.2.1.org/contrib/pngminus/makefile.std	Wed Mar 13 11:40:24 2002
+++ libpng-1.2.1/contrib/pngminus/makefile.std	Wed Mar 13 11:46:05 2002
@@ -19,13 +19,13 @@
 #ZINC = -I$(ZPATH)/include
 #ZLIB = -L$(ZPATH)/lib -lz
 #ZLIBS = $(ZPATH)/lib/libz.a
-ZINC = -I../../../zlib
-ZLIB = -L../../../zlib -lz
-ZLIBS = ../../../zlib/libz.a
+ZINC = 
+ZLIB = -lz
+ZLIBS = -lz
 
-CFLAGS=-O3 $(PNGINC) $(ZINC)
+CFLAGS=$(OPT_FLAGS) $(PNGINC) $(ZINC)
 LDFLAGS=$(PNGLIB) $(ZLIB)
-LDFLAGSS=$(PNGLIBS) $(ZLIBS)
+LDFLAGSS=$(LD_FLAGS) $(PNGLIBS) $(ZLIBS)
 C=.c
 O=.o
 L=.a
