diff -Nur libpng-1.0.15.orig/scripts/makefile.linux libpng-1.0.15/scripts/makefile.linux
--- libpng-1.0.15.orig/scripts/makefile.linux	Thu Jan  9 22:20:03 2003
+++ libpng-1.0.15/scripts/makefile.linux	Thu Jan  9 22:36:26 2003
@@ -3,8 +3,8 @@
 # Copyright (C) 1996, 1997 Andreas Dilger
 # For conditions of distribution and use, see copyright notice in png.h
 
-LIBNAME = libpng10
-PNGMAJ = 0
+LIBNAME = libpng
+PNGMAJ = 2
 PNGMIN = 1.0.15
 PNGVER = $(PNGMAJ).$(PNGMIN)
 
@@ -34,7 +34,7 @@
 CFLAGS= $(OPT_FLAGS) -Wall -funroll-loops \
 	$(ALIGN) # $(WARNMORE) -DPNG_DEBUG=5
 
-LDFLAGS=-L. -Wl,-rpath,. -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) -lpng10 -lz -lm
+LDFLAGS=-L. -Wl,-rpath,. -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) -lpng -lz -lm
 LDFLAGS_A=-L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) libpng.a -lz -lm
 
 RANLIB=ranlib
@@ -97,14 +97,13 @@
 	ln -sf $(LIBNAME).so.$(PNGVER) $(LIBNAME).so.$(PNGMAJ)
 
 $(LIBNAME).so.$(PNGVER): $(OBJSDLL)
-	$(CC) -shared -Wl,-soname,$(LIBNAME).so.$(PNGMAJ) \
+	$(CC) -shared -Wl,-soname,$(LIBNAME).so.$(PNGMAJ) -lz -lm \
 	-o $(LIBNAME).so.$(PNGVER) \
 	$(OBJSDLL)
 
-libpng.so.2.$(PNGMIN): $(OBJSDLL)
-	$(CC) -shared -Wl,-soname,libpng.so.2 \
-	-o libpng.so.2.$(PNGMIN) \
-	$(OBJSDLL)
+libpng10.so.0.$(PNGMIN): $(LIBNAME).so.$(PNGVER)
+	ln -sf $(LIBNAME).so.$(PNGVER) libpng10.so.0.$(PNGMIN)
+	ln -sf libpng10.so.0.$(PNGMIN) libpng10.so.0
 
 pngtest: pngtest.o $(LIBNAME).so
 	$(CC) -o pngtest $(CFLAGS) pngtest.o $(LDFLAGS)
@@ -123,42 +123,41 @@
 
 install-headers: png.h pngconf.h
 	-@if [ ! -d $(DI) ]; then mkdir $(DI); fi
-	-@if [ ! -d $(DI)/$(LIBNAME) ]; then mkdir $(DI)/$(LIBNAME); fi
-	cp png.h pngconf.h $(DI)/$(LIBNAME)
-	chmod 644 $(DI)/$(LIBNAME)/png.h $(DI)/$(LIBNAME)/pngconf.h
+	-@if [ ! -d $(DI)/libpng10 ]; then mkdir $(DI)/libpng10; fi
+	cp png.h pngconf.h $(DI)/libpng10
+	chmod 644 $(DI)/libpng10/png.h $(DI)/libpng10/pngconf.h
 	-@/bin/rm -f $(DI)/png.h $(DI)/pngconf.h
 	-@/bin/rm -f $(DI)/libpng
-	(cd $(DI); ln -sf $(LIBNAME) libpng; ln -sf $(LIBNAME)/* .)
+	(cd $(DI); ln -sf libpng10 libpng; ln -sf libpng10/* .)
 
 install-static: install-headers libpng.a
 	-@if [ ! -d $(DL) ]; then mkdir $(DL); fi
 	cp libpng.a $(DL)/$(LIBNAME).a
 	chmod 644 $(DL)/$(LIBNAME).a
-	-@/bin/rm -f $(DL)/libpng.a
-	(cd $(DL); ln -sf $(LIBNAME).a libpng.a)
+	-@/bin/rm -f $(DL)/libpng10.a
+	(cd $(DL); ln -sf $(LIBNAME).a libpng10.a)
 
 install-shared: install-headers $(LIBNAME).so.$(PNGVER) libpng.pc \
-	libpng.so.2.$(PNGMIN)
+	libpng10.so.0.$(PNGMIN)
 	-@if [ ! -d $(DL) ]; then mkdir $(DL); fi
 	-@/bin/rm -f $(DL)/$(LIBNAME).so.$(PNGVER)* $(DL)/$(LIBNAME).so
-	-@/bin/rm -f $(DL)/libpng.so
-	-@/bin/rm -f $(DL)/libpng.so.2
-	-@/bin/rm -f $(DL)/libpng.so.2.$(PNGMIN)*
+	-@/bin/rm -f $(DL)/libpng10.so
+	-@/bin/rm -f $(DL)/libpng10.so.0
+	-@/bin/rm -f $(DL)/libpng10.so.0.$(PNGMIN)*
 	cp $(LIBNAME).so.$(PNGVER) $(DL)
-	cp libpng.so.2.$(PNGMIN) $(DL)
 	chmod 755 $(DL)/$(LIBNAME).so.$(PNGVER)
-	chmod 755 $(DL)/libpng.so.2.$(PNGMIN)
 	(cd $(DL); \
-	ln -sf libpng.so.2.$(PNGMIN) libpng.so.2; \
-	ln -sf libpng.so.2 libpng.so; \
+	ln -sf $(LIBNAME).so.$(PNGVER) libpng10.so.0.$(PNGMIN); \
+	ln -sf libpng10.so.0.$(PNGMIN) libpng10.so.0; \
+	ln -sf libpng10.so.0 libpng10.so; \
 	ln -sf $(LIBNAME).so.$(PNGVER) $(LIBNAME).so.$(PNGMAJ); \
 	ln -sf $(LIBNAME).so.$(PNGMAJ) $(LIBNAME).so)
 	-@if [ ! -d $(DL)/pkgconfig ]; then mkdir $(DL)/pkgconfig; fi
 	-@/bin/rm -f $(DL)/pkgconfig/$(LIBNAME).pc
-	-@/bin/rm -f $(DL)/pkgconfig/libpng.pc
+	-@/bin/rm -f $(DL)/pkgconfig/libpng10.pc
 	cp libpng.pc $(DL)/pkgconfig/$(LIBNAME).pc
 	chmod 644 $(DL)/pkgconfig/$(LIBNAME).pc
-	(cd $(DL)/pkgconfig; ln -sf $(LIBNAME).pc libpng.pc)
+	(cd $(DL)/pkgconfig; ln -sf $(LIBNAME).pc libpng10.pc)
 
 install-man: libpng.3 libpngpf.3 png.5
 	-@if [ ! -d $(DM) ]; then mkdir $(DM); fi
@@ -178,7 +176,7 @@
 	-@/bin/rm -f $(DB)/$(LIBNAME)-config
 	cp libpng-config $(DB)/$(LIBNAME)-config
 	chmod 755 $(DB)/$(LIBNAME)-config
-	(cd $(DB); ln -sf $(LIBNAME)-config libpng-config)
+	(cd $(DB); ln -sf $(LIBNAME)-config libpng10-config)
 
 install: install-static install-shared install-man install-config
 
