--- libpng-1.2.6rc1/pngpread.c.orig	2004-08-04 13:34:47.000000000 +0200
+++ libpng-1.2.6rc1/pngpread.c	2004-08-07 13:52:13.320796920 +0200
@@ -1009,7 +1009,7 @@
             png_pass_inc[png_ptr->pass];
 
          png_ptr->irowbytes = PNG_ROWBYTES(png_ptr->pixel_depth,
-            png_ptr->iwidth);
+            png_ptr->iwidth) + 1;
 
          if (png_ptr->transformations & PNG_INTERLACE)
             break;
