--- libpng-1.2.6beta2/pnggccrd.c.orig	2002-11-01 02:53:08.000000000 +0100
+++ libpng-1.2.6beta2/pnggccrd.c	2004-03-01 15:17:55.807326104 +0100
@@ -5036,7 +5036,10 @@
         "1" (prev_row),         // esi
         "2" (row)               // edi
 
-      : "%eax", "%ebx", "%ecx"            // clobber list (no input regs!)
+      : "%eax", "%ecx"          // clobber list (no input regs!)
+#ifndef __PIC__
+      , "%ebx"
+#endif
 
 #if 0  /* MMX regs (%mm0, etc.) not supported by gcc 2.7.2.3 or egcs 1.1 */
       , "%mm0", "%mm1", "%mm2", "%mm3"
