--- libpng-1.2.21/scripts/makefile.gcmmx~	2007-10-08 19:52:12.000000000 +0200
+++ libpng-1.2.21/scripts/makefile.gcmmx	2007-10-08 19:55:00.000000000 +0200
@@ -70,8 +70,8 @@
 #	-m486 -malign-double -ffast-math \
 #	$(ALIGN) -funroll-loops -funroll-all-loops -fomit-frame-pointer 
 
-LDFLAGS=-L. -Wl,-rpath,. -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) -lpng12 -lz -lm
-LDFLAGS_A=-L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) libpng.a -lz -lm
+LDFLAGS=-L. -lpng12 -lz -lm
+LDFLAGS_A=libpng.a -lz -lm
 
 
 INCPATH=$(prefix)/include
@@ -124,7 +124,7 @@
 	echo I_opts=\"-I$(INCPATH)/$(LIBNAME)\"; \
 	echo cppflags=\"-DPNG_THREAD_UNSAFE_OK \"; \
 	echo L_opts=\"-L$(LIBPATH)\"; \
-	echo R_opts=\"-Wl,-rpath,$(LIBPATH)\"; \
+	echo R_opts=\"\"; \
 	echo libs=\"-lpng12 -lz -lm\"; \
 	cat scripts/libpng-config-body.in ) > libpng-config
 	chmod +x libpng-config
@@ -238,7 +238,6 @@
 test-installed:
 	$(CC) -I$(ZLIBINC) \
 	   `$(BINPATH)/$(LIBNAME)-config --cflags` pngtest.c \
-	   -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) \
 	   -o pngtesti `$(BINPATH)/$(LIBNAME)-config --ldflags`
 	./pngtesti pngtest.png
 
--- libpng-1.2.21/scripts/makefile.linux~	2007-10-08 19:52:12.000000000 +0200
+++ libpng-1.2.21/scripts/makefile.linux	2007-10-08 19:56:43.000000000 +0200
@@ -50,8 +50,8 @@
 CFLAGS=-I$(ZLIBINC) -Wall $(OPT_FLAGS) -funroll-loops \
 	$(ALIGN) # $(WARNMORE) -g -DPNG_DEBUG=5
 
-LDFLAGS=-L. -Wl,-rpath,. -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) -lpng12 -lz -lm
-LDFLAGS_A=-L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) libpng.a -lz -lm
+LDFLAGS=-L. -lpng12 -lz -lm
+LDFLAGS_A=libpng.a -lz -lm
 
 INCPATH=$(prefix)/include
 LIBPATH=$(exec_prefix)/lib
@@ -102,7 +102,7 @@
 	echo prefix=\"$(prefix)\"; \
 	echo I_opts=\"-I$(INCPATH)/$(LIBNAME)\"; \
 	echo L_opts=\"-L$(LIBPATH)\"; \
-	echo R_opts=\"-Wl,-rpath,$(LIBPATH)\"; \
+	echo R_opts=\"\"; \
 	echo libs=\"-lpng12 -lz -lm\"; \
 	cat scripts/libpng-config-body.in ) > libpng-config
 	chmod +x libpng-config
@@ -214,7 +214,6 @@
 test-installed:
 	$(CC) -I$(ZLIBINC) \
 	   `$(BINPATH)/$(LIBNAME)-config --cflags` pngtest.c \
-	   -L$(ZLIBLIB) -Wl,-rpath,$(ZLIBLIB) \
 	   -o pngtesti `$(BINPATH)/$(LIBNAME)-config --ldflags`
 	./pngtesti pngtest.png
 
